llvm/lib/Transforms/Instrumentation/MemProfUse.cpp

```diff
@@ -361,6 +361,74 @@ static void addVPMetadata(Module &M, Instruction &I,
   }
 }
 
+static void
+handleAllocSite(Instruction &I, CallBase *CI,
```

Comment:
Longer term, we should create a new class to encapulate the matching functionality so that we can internalize the state being passed around and make these public member functions that can be unit tested.